app/models/study_session.rb

```diff
@@ -0,0 +1,5 @@
+class StudySession < ActiveRecord::Base
+	has_and_belongs_to_many :time_availabilitys, as: :time_availabilityable 
```

Comment:
Can this just be has_one?

Reply:
It had the same issue as adding a time availability to a tutor and a tutee (when we add ta_object1 for 1 session, it was taken off from another session). We have `has_and_belongs_to_many` to allow multiple pairs of tutors and tutees to meet at a single time. i.e. tutee1 and tutor1 can meet at 8am on Monday and so can tutee2 and tutor2. In practice however, we never add more than 1 time_availability to a session for a particular pair of tutor and tutee.

Reply:
So to confirm, even though there is only one time availability per session, this still has to be plural?

Reply:
Yes, that's correct. It's to resolve the problem described in the previous comment

Reply:
I think this is ready to be pushed if Juno agrees. Do you want to wait for him to sign off too?